app/main.py

```diff
@@ -1,34 +1,119 @@
+class IncorrectNumberOfShips(Exception):
+    def __str__(self) -> str:
+        return "The total number of the ships should be 10"
+
+
+class IncorrectShipCoordinates(Exception):
+    def __str__(self) -> str:
+        return ("The ends of the ship must lie on the same"
+                " line and its length must not exceed 4 cells")
+
+
+class IncorrectSetOfShips(Exception):
+    def __str__(self) -> str:
+        return ("there should be 4 single-deck, 3 double-deck,"
+                " 2 three-deck and 1 four-deck ship")
+
+
+class IncorrectPositionOfShips(Exception):
+    def __str__(self) -> str:
+        return ("Ships shouldn't be located in the neighboring cells"
+                " (even if cells are neighbors by diagonal).")
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.decks = []
+        if start[0] == end[0]:
+            for index in range(start[1], end[1] + 1):
+                self.decks.append(Deck(start[0], index))
+        else:
+            for index in range(start[0], end[0] + 1):
+                self.decks.append(Deck(index, start[1]))
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
```

Comment:
What if `get_deck` return `None`?

Reply:
this is not possible due to the implementation of the algorithm. in the Battleship class, in the self.field field, I enter only non-empty cells that are parts of ships. Further, before each get_deck call, I check if the given coordinates are in self.field